core/src/main/java/io/lionweb/lioncore/java/serialization/NodePopulator.java

```diff
@@ -0,0 +1,116 @@
+package io.lionweb.lioncore.java.serialization;
+
+import io.lionweb.lioncore.java.api.ClassifierInstanceResolver;
+import io.lionweb.lioncore.java.language.Classifier;
+import io.lionweb.lioncore.java.language.Containment;
+import io.lionweb.lioncore.java.language.Reference;
+import io.lionweb.lioncore.java.model.ClassifierInstance;
+import io.lionweb.lioncore.java.model.Node;
+import io.lionweb.lioncore.java.model.ReferenceValue;
+import io.lionweb.lioncore.java.serialization.data.SerializedClassifierInstance;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * This helper class take care of populating containments and references of a node, while we are
+ * deserializing it.
+ */
+class NodePopulator {
+  private JsonSerialization jsonSerialization;
```

Comment:
`final`?

Reply:
Making all these fields final in follow up commit